### PR TITLE
Add test cases to verify that error responses contain status code

### LINKED ISSUE #912: Error codes for failed repo webhook requests are "Unknown"
`Repositories.ListHooks(ctx, owner, repository, opts)` returns hooks, response, and error. If response failed with codes such as `404` or `403`, this is not reflected in the error code. Using `errors.Code(err)` when `response` returns an error HTTP status code will simply return `codes.Unknown`, when it should really be `codes.PermissionDenied` or `codes.NotFound`. 

Propose a change where we map HTTP Response status codes (if error) into the error object's error code.

### PR DESCRIPTION
**Pull request by AGMETEOR**
Fixes #912

### CODE REVIEW

**Comment by AGMETEOR** [comment]
Hi @gmlewis , I hope you has a great weekend. Here's the PR for issue 912

**Review by cpanato — APPROVED** [review]
LGTM

File: github/github_test.go [hunk 1]
```diff
@@ -216,6 +216,32 @@ func testNewRequestAndDoFailure(t *testing.T, methodName string, client *Client,
 	}
 }
 
+// Test that all error response types contain the status code
```
Comment (gmlewis) [inline github/github_test.go:219]:
```suggestion
// Test that all error response types contain the status code.
```

**Review by gmlewis — APPROVED** [review]
Thank you, @AGMETEOR !
LGTM.
Thank you, also, @cpanato for the review!  We appreciate it.
Merging.

**Pull request merged**